crypto/conf/conf_sap.c

```diff
@@ -35,7 +31,7 @@ void OPENSSL_config(const char *appname)
 
     memset(&settings, 0, sizeof(settings));
     if (appname != NULL)
-        settings.appname = strdup(appname);
+        settings.appname = OPENSSL_strdup(appname);
     settings.flags = DEFAULT_CONF_MFLAGS;
     OPENSSL_init_crypto(OPENSSL_INIT_LOAD_CONFIG, &settings);
```

Comment:
this creates just a memroy leak, so the strdup should be remove here.

Reply:
The memory leak was there already with strdup(), wasn't it? It is a one-off memleak.

Reply:
yeah correct, I've remembered I noticed it already in the past...

Reply:
it is a one off, but at the same time, it seems pretty easy to fix.  since the settings variable is declared on the stack, you can just do an OPENSSL_free on settings.appname after the call to OPENSSL_init_crypto

Reply:
sure, but my point is that it is perfectly safe to use the original pointer, because OPENSSL_init_crypto
will not modify it, and use it just for the duration of the call.